app/code/Magento/Catalog/Helper/Product/ProductList.php

```diff
@@ -97,6 +100,12 @@ public function getDefaultViewMode($options = [])
      */
     public function getDefaultSortField()
     {
+        $currentCategory = $this->coreRegistry->registry('current_category');
+        if ($currentCategory) {
+            $currentCategorySortBy = $currentCategory->getDefaultSortBy();
```

Comment:
Is there any reason for setting this as a variable rather than `return $currentCategory->getDefaultSortBy();`. There is no real problem with setting the variable but if it is not needed then I would recommend a quick return.